### PR TITLE
add an example

### PR DESCRIPTION
**Pull request by orefalo**
adds a simple example

### CODE REVIEW

File: example/index.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                access: "readWrite"
+              }
+            ]
+          }
+        }
+      })
+    }
+  ],
+  getExecutionOptions: ({ context }) => ({ authRoles: ["users"] }),
+//   getOperationIdentifier: ({ context }) => context as object // each operation is executed with an unique context object
```
Comment (Yogu) [inline example/index.ts]: I would un-comment this; you pass the request as context, so this should be fine

File: example/index.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+try {
+  const schema = project.createSchema(db)
+  db.updateSchema(project.getModel()) // create missing collections
+  const server = new ApolloServer({
+    schema,
+    context: ({ req }) => req // pass request as context so we have a unique context object for each operation
+  })
+  server.listen(4000, () => console.log("Server is running on http://localhost:4000/"))
+} catch (error) {
+  console.log("Did you create the 'test' database?")
```
Comment (Yogu) [inline example/index.ts]: wouldn't swallow all errors, even in a small test. You could add the log output in addition to `error.stack` though.

**Comment by Yogu** [comment]
Thank's alot! 💚

File: example/index.ts (hunk 1 above)
Comment (orefalo) [reply example/index.ts]: right, this is for version 1.0.5 - getOperationIdentifier is not there yet

Comment (orefalo) [reply example/index.ts]: I take this back.. I get this TS error:
```
Argument of type '{ sources: { name: string; body: string; }[]; getExecutionOptions: ({ context }: ExecutionOptionsCallbackArgs) => { authRoles: string[]; }; getOperationIdentifier: ({ context }: { context: any; }) => object; }' is not assignable to parameter of type 'ProjectConfig | SourceLike[]'.
  Object literal may only specify known properties, and 'getOperationIdentifier' does not exist in type 'ProjectConfig | SourceLike[]'.ts(2345)
```

Comment (orefalo) [reply example/index.ts]: added a ts-ignore to get around the typing issue

**Comment by orefalo** [comment]
closing. Will make a clean resubmission on master

**Comment by Yogu** [comment]
Ah ok, otherwise I would have just squashed the whole PR into one commit.

File: example/index.ts (hunk 1 above)
Comment (Yogu) [reply example/index.ts]: ah of course, I forgot.

**Comment by itsezc** [comment]
@orefalo Would you be kind enough to commit examples with Apollo Federation and Authentication, would be very useful!

**Comment by orefalo** [comment]
> @orefalo Would you be kind enough to commit examples with Apollo Federation and Authentication, would be very useful!

I don't have time, sorry.

FYI: Federation is not compatible with cruddl. Authentication with gql&oauth are pretty common nowadays.

**Pull request closed without merging**